### ISSUE TITLE
Trouble with Grid Editor

### ISSUE THREAD

**Issue by IanMayo** [debrief/debrief#1263]
A reviewer has found a problem with multi-selecting Sensor items in the Grid Editor (video attached).

But, I couldn't get as far enough to reproduce the problem - due to a problem related to our "Display Name" changes.

Here are the steps:
1. Open Debrief, no plot open
2. Load *sen_tracks.rep*
3. Load *sen_ssk_sensor.dsf*  (accept defaults)
4. Make the SSK sensor data visible
5. Open the grid editor
6. Select one of the blocks of sensor data  *New_SSK_BB*

A warning will then open, indicating that it can't find the relevant method.  It's trying to retrieve the attribute using the **Display Name** not the **Name**.

Once you're on top of that - we can investigate the interpolation error recorded in the video.

[Interpolation Error 12-1.mp4 | uploaded via ZenHub](https://files.zenhub.io/55717775b282af931c502643)

**Comment by pecko** [comment]
The PR: https://github.com/debrief/debrief/pull/1266

**Comment by IanMayo** [comment]
@pecko  - does it fix the bug in the video
That bug appears to be related to arithmetic. I'm happy to look at it if it looks complex.

**Comment by IanMayo** [comment]
I've tried the PR - it works fine.  I'll merge it now, since it's a significant problem.

But, I'll leave this Issue open, to investigate the Grid Editor selection problem.  I don't experience the error at my machine, so we may need to direct the tester to get some diagnostics (error messages for us).

**Comment by pecko** [comment]
> does it fix the bug in the video

I also can't reproduce it, but if it happens after fixing #1161 we probably solve it.

**Comment by IanMayo** [comment]
No, I don't think it will be fixed. He's using a pre-property descriptor
change version of Debrief.

**Comment by pecko** [comment]
OK. I have tried only the case you described (seen video, but not deeply). Will investigate slightly today later or tomorrow. We will probably have to wait for user's log.

**Comment by pecko** [comment]
There are two classes where an exception from video can happen (both are in org.mwc.cmap.legacy/libs/flanagan.jar):

- flanagan.interpolation.CubicSpline.interpolate(double)
- flanagan.interpolation.LinearInterpolation.interpolate(double)

The engine can't do interpolation in any cases.
We can catch these exceptions, but not sure what to do in this case:

- show a dialog to a user with better explanation
or
- just log an exception
or
- do nothing

@IanMayo  WDYT?

BTW
We could also change global potentialRoundingError (potential rounding error used in checking whether a value lies within the interpolation bounds) which is 5e-15 by default, but it couldn't eliminate a problem (maybe the problem would happen more rarely, but it would happen)

**Comment by IanMayo** [comment]
[sorry Peco, didn't get around to this - will try again tomorrow]

**Comment by IanMayo** [comment]
Hmm, I can't reproduce it.  I wonder if it's a time zone issue?  (Russ is in the Far East).

No, I don't think it can be, since the other three cells get selected ok.

In the video footage, it's reporting  that the x value is beyond the range of the 

I don't think that changing the rounding error will fix it.  I think that 5e-15 is a lot smaller than the 1.xe12 that we're working with.

I'm keen to get to the bottom of it.  I'll get back to Russ - to experiment a little more.

**Comment by pecko** [comment]
I also don't think that it's a time zone issue because we set UTC (GMT+0) time zone explicitelly at level of JVM (when starting Debrief).

Maybe Russ changed a values anyhow, but it seems that he has the same values as me.

**Comment by IanMayo** [comment]
Yes, he has the same values as me - that is why I was looking for something that's different (like time zone).

I'll ask him if it's reproducible, then provide some instructions in providing an Error Log.
